stream.go

```diff
@@ -65,6 +66,12 @@ func (dec *Decoder) NumBytesRead() int {
 	return dec.bytesRead
 }
 
+// Buffered returns a reader for data remaining in Decoder's buffer.
+// Returned reader is valid until the next call to Decode or Skip.
+func (dec *Decoder) Buffered() io.Reader {
+	return bytes.NewReader(dec.buf[dec.off:])
+}
```

Comment:
Thanks for adding this and making it match encoding/json.

Are you going to add `Decoder.Buffered` to README in separate PR?

Reply:
Good catch!  I'll open separate PR to update README because I already have unrelated changes to README in progress.